ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -868,15 +868,21 @@ def node_sid(self):
                 return self.id + node_sid_base
         return None
 
+    @cached_property
+    def underlay_ipv6(self):
+        if self.underlay_router is True:
+            return get(self._hostvars, "underlay_ipv6")
+        return None
```

Comment:
According to documentation, there is a default value for `underlay_ipv6` which is `False` so should we really return `None` here?

Reply:
Strictly speaking, this is an internal var which is not necessarily following the input, but I understand your point. The None value will be removed from the switch facts before output, so that is why we have `is arista.avd.defined(true)` in the templates. If we change this to False, we could also simplify the templates to just use `is true`

Reply:
ok I see - so the template will be interpreting the absence of value as `false` in the end as is documented. 
To some extent it means we are never explicitly setting the documented default value just interpreting it implicitly and it may come back to bite us someday but in this specific scenario I think it is quite safe to stay like this.

Another downside may come the day we decide to add type hint in the code though we can get out with `Optional` convenience definition from PEP-0484.